doc/qos_overview.md

```diff
@@ -422,3 +429,278 @@ markings. The same interface has an egress scheduler policy applied to it.
   }
 }
 ```
+
+### Example 2: Ingress Classification with Ingress Scheduling (Policer) on a VOQ device
+
+The example QoS configuration below shows the configuration of an interface,
+assumed to be facing a customer which has ingress classification based on
+the next-hop group the packet is being sent to.  The same interface has an
+ingress scheduler policy applied to it which implements an
+`ONE_RATE_TWO_COLOR` policer.
+
+In this scenario, the device has a VOQ architecture is does not have hardware
+or software to implement in ingress queue.  To satisfy the OC schema
+requirements, a dummy or "fake" queue is created for the ingress side of the
+pipeline.  Note, an egress queue could still be defined on the egress side,
+but it is not included here for simplication.
+
+```json
+{
+  #
+  # A classifer is created to match packets belonging to certain
+  # next-hop-groups and map them either forwarding-group input_dest_A or
+  # input_dest_B
+  "openconfig-qos": {
+    "classifers": [
+      {
+        "classifer": "“dest_A”",
+        "config": {
+          "name": "“dest_A”"
+        },
+        "terms": [
+          {
+            "term": null,
```

Comment:
Not sure what this `term` field that is set to `null` is -- the JSON here seems incorrect:

```json
{
  "qos": {
    "classifiers": {
      "classifier": [
        {
          "name": "Dest_A",
          "config": {
            "name": "Dest_A"
          },
          "terms": {
            "term": [
              {
                "id": "foo"
              }
            ]
          }
        }
      ]
    }
  }
}
```

I suggest that this is generated using pyangbind, ygot, or your-favourite-YANG-thingy -- you can also write it by hand and validate it using something like [`govalidate`](https://github.com/openconfig/oc-tools/blob/master/validate/govalidate/main.go) (although this needs some maintenance to add some new modules and update the docs).

Reply:
FYI, the above JSON validates: 

```
[2:47] robjs@aardvark:oc-tools/validate/govalidate   master ✔                                                                       242d ⚑ ◒  ⍉
▶ ./govalidate -fn foo.json && echo $? 
0
```